webapp/channels/src/components/drafts/draft_row.tsx

```diff
@@ -177,6 +164,46 @@ function DraftRow({
         dispatch(removeDraft(key, channelId, rootId));
     }, [dispatch, channelId, rootId]);
 
+    // afterSubmit is exclusively used for deleting the draft after
+    // successfully creating a scheduled post from it.
+    // This is not used in any other draft flow such as sending it, or editing it etc.
+    const afterSubmit = useCallback((response: SubmitPostReturnType) => {
+        if (isBeingScheduled.current && response.created && !response.error) {
+            handleOnDelete();
+            isBeingScheduled.current = false;
+        }
+
+        if (isScheduledPostBeingSent.current && response.created && !response.error) {
+            dispatch(deleteScheduledPost((item as ScheduledPost).id, connectionId));
+            isScheduledPostBeingSent.current = false;
+        }
+    }, [connectionId, dispatch, handleOnDelete, item]);
+
+    // TODO LOL verify the types and handled it better
+    const {onSubmitCheck: prioritySubmitCheck} = usePriority(item as any, noop, noop, false);
+    const [handleOnSend] = useSubmit(
+        item as any,
+        postError,
+        channelId,
+        rootId,
+        serverError,
+        mockLastBlurAt,
+        noop,
+        setServerError,
+        noop,
+        noop,
+        prioritySubmitCheck,
+        goToMessage,
+        afterSubmit,
+        true,
+    );
+
+    const onScheduleDraft = useCallback(async (scheduledAt: number): Promise<{error?: string}> => {
+        isBeingScheduled.current = true;
+        await handleOnSend(item as PostDraft, {scheduled_at: scheduledAt});
+        return Promise.resolve({});
```

Comment:
Isn't this supposed to return an error if any?

Reply:
This return here is used by the scheduled time picker modal to display errors, but in case of scheduling a draft, I;'m displaying the error in draft row itself as the useSubmit hook doesn't directly return the error and its a lot of refactoring to make it do so.